### PR TITLE
Use mono wasm method to invoke entry point

### PR DESCRIPTION
**Pull request by pranavkm**
Taking advantage of https://github.com/mono/mono/commit/cdf8b839aef8068f601bdbed5b3fd7ec4acc462f. My hope's to build upon this to simplify \ remove BootJsonWriter

### CODE REVIEW

File: src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts [hunk 1]
```diff
@@ -39,21 +40,8 @@ export const monoPlatform: Platform = {
...
-    const typeFullName = entrypointSegments[0];
-    const methodName = entrypointSegments[1];
-    const lastDot = typeFullName.lastIndexOf('.');
-    const namespace = lastDot > -1 ? typeFullName.substring(0, lastDot) : '';
-    const typeShortName = lastDot > -1 ? typeFullName.substring(lastDot + 1) : typeFullName;
-
-    const entryPointMethodHandle = monoPlatform.findMethod(assemblyName, namespace, typeShortName, methodName);
-    monoPlatform.callMethod(entryPointMethodHandle, null, args);
+  callEntryPoint: function callEntryPoint(assemblyName: string): System_Object {
+    return mono_call_assembly_entry_point(assemblyName, []);
```
Comment (pranavkm) [inline src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts]: Super unfamiliar with using Mono WASM. Do we need to "read" the returned value? `callMethod` does some extra work to read the exception string for instance.

Comment (SteveSandersonMS) [reply src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts]: > callMethod does some extra work to read the exception string for instance.

`mono_wasm_invoke_method` takes an `int*` into which it writes a flag that signals whether the return value is an exception or not.

It looks like `mono_call_assembly_entry_point` does not have any comparable way of signaling that the return value is an exception, so it doesn't look to me like there's anything else for you to do.

Can you check what the behavior is, before and after your change, if the entrypoint throws? If we're regressing (previously logged the error to the console, but now just swallow it), then we might not want to do this. As long as we do still write out any entrypoint exception to the browser console then I think we're OK here.

**Review by SteveSandersonMS — APPROVED** [review]
This definitely looks like an improvement! As long as it's not regressing the exception-handling behavior, then 👍 from me.

File: src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts (hunk 1 above)
Comment (pranavkm) [reply src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts]: Yup, it prints the exception to the console. I'm following up on the behavior of the args array being corrupt

**Pull request merged**